src/ztc/api/tests/test_zaken.py

```diff
@@ -138,6 +139,9 @@ def test_create_zaaktype(self):
         besluittype = BesluitTypeFactory.create(catalogus=self.catalogus)
         besluittype_url = get_operation_url("besluittype_read", uuid=besluittype.uuid)
 
+        deelzaaktype1 = ZaakTypeFactory.create(catalogus=self.catalogus, concept=False)
```

Comment:
is it allowed to add a published ZaakType to another ZaakType? And vice versa, is it allowed to add a ZaakType to a published ZaakType?

Reply:
@joeribekker ?

Reply:
I think the most practical approach is the following 2 rules:

* You can add a published object to a non-published object.
* You cannot add a published or non-published object to a published object.

So, in context of your questions:

> is it allowed to add a published ZaakType to another ZaakType?

Yes, as long as the 2nd ZaakType is not published.

> And vice versa, is it allowed to add a ZaakType to a published ZaakType?

No.

Reply:
Yeah, I had a similar reasoning - since you are currently editing a `ZaakType` (create or update), it is not published yet - you can change anything you want. So, you can add `deelzaaktypen` that are both published and not published.

However, I think that _when_ you publish the zaaktype with the `deelzaaktypen`, _then_ you need to validate that the `deelzaaktypen` are published.

We only keep a reference from hoofdzaaktype to deelzaaktypen and not the other way around, because that would lead to cyclic dependencies again.